artiq/coredevice/comm_moninj.py

```diff
@@ -32,7 +32,12 @@ def __init__(self, monitor_cb, injection_status_cb, disconnect_cb=None):
         self._receive_task = None
 
     async def connect(self, host, port=1383):
+        if self._writer:
+            await self.close()
+
         self._reader, self._writer = await asyncio.open_connection(host, port)
+        set_keepalive(self._writer.transport._sock, 1, 1, 3)
```

Comment:
hack?

Reply:
although it works, yes this is a hack, since _sock is a private field of BaseTransport, which you shouldn't access outside of the class.